packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -58,42 +58,43 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
     }
 
     public async summarize(fullTree: boolean, trackState: boolean = true): Promise<IContextSummarizeResult> {
-        /**
-         * If trackState is true, call parent summarizer node to get the summary since it tracks the state from summary.
-         * It may update the GC nodes cache if data has changed since last summary.
-         * If trackState is false, get the summary directly from the summarizeInternal method which will update the
-         * GC nodes cache.
-         */
         if (trackState) {
             const summarizeResult = await super.summarize(fullTree);
-
-            // If this is the first time we are summarizing and nothing has changed since the last summary, we would
-            // not have updated the GC nodes cache. So, we need to get its initial value.
-            if (this.gcNodes === undefined) {
-                this.gcNodes = this.getInitialGCNodesFn
-                    ? await this.getInitialGCNodesFn()
-                    : [];
-            }
-
             return {
                 ...summarizeResult,
-                gcNodes: cloneGCNodes(this.gcNodes),
+                gcNodes: [],
```

Comment:
I've sent you an IM with more thoughts on this. But I believe it has to return a set of actual nodes that are left after GC was run. This info needs to be used on future GCs/summarization runs if this data store is not going to be loaded due to delay-loading.

Reply:
As per our discussion, I will keep gcNodes in summarize for now. When we implement GC at DDS / sub-DDS level, we can revisit this and decide then if we want to remove it.

I have added the following commit that updates summarize to return IGCData instead of gcNodes to make it consistent with getGCData API:
https://github.com/microsoft/FluidFramework/pull/4538/commits/63c52bcc9c01ca2cd4dccfcc30570dd3eac1cc40
I wanted to show this in a separate PR but that seems to be rather difficult without merging this first :-)